### DISCUSSION TITLE
How to use ShadowRoot elements effectively using typescript

### DISCUSSION THREAD

**Discussion by Arthanarieaswaran** [codeceptjs/CodeceptJS#3695] [Q&A] [answered]
I am trying to use { shadow } object in the locator, it is giving the below error in TYPESCRIPT

<img width="672" alt="image" src="https://github.com/codeceptjs/CodeceptJS/assets/6602970/4a3cf7ef-a475-48cd-829a-d2105d918f11">

Also how to use, XPath and cssSelectors (complicated identifier) in the shadow root

**Comment by kobenguyent** [comment, marked as answer]
shall be there in the new release https://github.com/codeceptjs/CodeceptJS/pull/3602
